src/common/com/intellij/plugins/haxe/model/HaxeClassModel.java

```diff
@@ -204,6 +211,85 @@ public SpecificHaxeClassReference getUnderlyingClassReference(HaxeGenericResolve
     return types;
   }
 
+  public List<SpecificHaxeClassReference> getCastableToTypesList(SpecificHaxeClassReference sourceType) {
+    if (!isAbstract()) return Collections.emptyList();
+    List<HaxeMethodModel> methodsWithMetadata = getCastToMethods();
+
+    return  methodsWithMetadata.stream()
+      .filter(methodModel -> castMethodAcceptsSource(sourceType, methodModel))
+      .map(m -> SetSpecificsConstraints(m, getReturnType(m)))
+      .collect(toList());
+  }
+
+
+  private boolean castMethodAcceptsSource(SpecificHaxeClassReference reference, HaxeMethodModel methodModel) {
+    SpecificHaxeClassReference parameter = getTypeOfFirstParameter(methodModel);
+    //implicit cast methods seems to accept both parameter-less methods and single parameter methods
+    if (parameter == null) return true; // if no param then "this" is  the  input  and will always be compatible.
+    SpecificHaxeClassReference parameterWithRealRealSpecifics = SetSpecificsConstraints(methodModel, parameter);
+
+    if(reference.isAbstract()) {
+      SpecificHaxeClassReference underlying = reference.getHaxeClassModel().getUnderlyingClassReference(reference.getGenericResolver());
+      if(canAssignToFrom(parameterWithRealRealSpecifics, underlying, false)) return true;
+    }
+
+    return canAssignToFrom(parameterWithRealRealSpecifics, reference, false);
+  }
+
+  public List<SpecificHaxeClassReference> getImplicitCastTypesList(SpecificHaxeClassReference targetType) {
+    if (!isAbstract()) return Collections.emptyList();
```

Comment:
Only abstracts??

Reply:
Its my understanding from reading the Manual that implicit cast only works on Abstracts
```
Implicit Casts

Unlike classes, abstracts allow defining implicit casts. There are two kinds of implicit casts:

    Direct: Allows direct casting of the abstract type to or from another type. This is defined by adding to and from rules to the abstract type and is only allowed for types which unify with the underlying type of the abstract.
    Class field: Allows casting via calls to special cast functions. These functions are defined using @:to and @:from metadata. This kind of cast is allowed for all types.
```